en/docs/develop-components/configure-endpoints-body.md

```diff
@@ -105,6 +105,115 @@ endpoints:
 
+### Apply enhancements to component configurations
+
+The `component-config.yaml` file allows you to apply enhancements to component configurations.
+
+!!! note "Beta release"
+      - The current version of the `component-config.yaml` file is considered stable. However, it is important to note that the configuration schema may undergo changes and improvements based on user feedback. 
+      - Support for the current schema will remain even when new versions are introduced.
+
+The `component-config.yaml` file complements the `endpoints.yaml` file and allows you to define configurations for both inbound and outbound connections.
+
+- **Inbound configurations:** This configuration section is for you to define inbound connections, specifically for endpoints. It aligns seamlessly with the existing endpoint object schema.
+
+- **Outbound configurations:** This configuration section is for you to specify outbound connection details, including service connections. The Choreo internal marketplace simplifies creating connections with existing services.  To learn more about Choreo Marketplace, see [Choreo Marketplace](https://wso2.com/choreo/docs/choreo-concepts/choreo-marketplace/#choreo-marketplace).
+
+!!! note
+    - If both `component-config.yaml` and `endpoints.yaml` are defined in the `.choreo` path, the `component-config.yaml` file takes priority.
+    - Outbound connections are not supported for deprecated components and WSO2 MI buildpack components.
+
+### Learn the `component-config.yaml` file
+
+The `component-config.yaml` file has a specific structure and contains the following details:
+
+| Field                | Required     | Description                                                                           |
+|----------------------|--------------|---------------------------------------------------------------------------------------|
+| **apiVersion**       | Required     | The version of the `component-config.yaml` file defaults to `core.choreo.dev/v1beta1`.|
+| **kind**             | Required     | The resource type of the file defaults to `ComponentConfig`.                          |
+| **spec.inbound**     | Optional     | The list of inbound connection configurations.                                        |
+| **spec.outbound**    | Optional     | The list of outbound connection configurations.                                       |
+
+
+#### Inbound connection configurations (spec.inbound)
+
+The `spec.inbound` section simplifies the configuration of inbound connections, allowing for the inclusion of a list of endpoints. It fully supports the existing endpoints schema structure. Refer to the [endpoints schema in the documentation](#learn-the-endpointsyaml-file) for further details.
+
+#### Outbound connection configurations (spec.outbound)
+
+Within the `spec.outbound` section, you can define `serviceReferences`, specifying the service references generated in the internal API marketplace when creating a service connection.
```

Comment:
Shall we say:
In the `spec.outbound` section, you can define `serviceReferences` by specifying the service references generated in the internal API marketplace when creating a service connection.

Reply:
Noted